src/main/java/Duke.java

```diff
@@ -1,10 +1,49 @@
+import Duke.commands.Command;
+import Duke.exceptions.DukeException;
+import Duke.parser.Parser;
+import Duke.storage.Storage;
+import Duke.tasks.TaskList;
+import Duke.ui.UI;
+
+import java.time.LocalDate;
+
 public class Duke {
+
+    private final Storage storage;
+    private TaskList tasks;
+    private final UI ui;
+
+    public Duke(String filePath) {
+        ui = new UI();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() {
```

Comment:
I really like how all the detailed functionalities of Duke.run() are abstracted further into classes.